### PR TITLE
Instrumentation to monitor number of capture data bytes sent

### PR DESCRIPTION
**Pull request by ronaldfw**
Add a small amount of instrumentation in OrbitService to monitor the
number of bytes of capture data sent during a capture. The average
number of bytes per CaptureEvent message sent is already recorded for
every batch of events sent and shown as a live graph when --devmode
is enabled.

Tested: Manual test.
Bug: http://b/175383580

### CODE REVIEW

File: OrbitService/CaptureServiceImpl.cpp [hunk 1]
```diff
@@ -91,30 +91,48 @@ class GrpcCaptureEventSender final : public CaptureEventSender {
     CHECK(reader_writer_ != nullptr);
   }
 
+  ~GrpcCaptureEventSender() {
+    LOG("Total number of events sent: %lu", total_number_of_events_sent_);
+    LOG("Total number of bytes sent: %lu", total_number_of_bytes_sent_);
+    LOG("Average number of bytes per event: %.2f",
+        static_cast<float>(total_number_of_bytes_sent_) / total_number_of_events_sent_);
```
Comment (dimitry-) [inline OrbitService/CaptureServiceImpl.cpp:98]: I do not really know: is it ok to divide float to 0 in c++?

**Review by dimitry- — APPROVED** [review]
one question, otherwise lgtm

File: OrbitService/CaptureServiceImpl.cpp (hunk 1 above)
Comment (ronaldfw) [reply OrbitService/CaptureServiceImpl.cpp:98]: Good catch, that would cause a divide by zero exception. Will fix.

Comment (ronaldfw) [reply OrbitService/CaptureServiceImpl.cpp]: Done.

File: OrbitService/CaptureServiceImpl.cpp [hunk 2]
```diff
@@ -91,30 +91,52 @@ class GrpcCaptureEventSender final : public CaptureEventSender {
     CHECK(reader_writer_ != nullptr);
   }
 
+  ~GrpcCaptureEventSender() {
```
Comment (dpallotti) [inline OrbitService/CaptureServiceImpl.cpp:94]: Could you add `override` to this (clang-tidy)?
```suggestion
  ~GrpcCaptureEventSender() override {
```

File: OrbitService/CaptureServiceImpl.cpp [hunk 3]
```diff
@@ -91,30 +91,52 @@ class GrpcCaptureEventSender final : public CaptureEventSender {
...
+    LOG("Total number of events sent: %lu", total_number_of_events_sent_);
+    LOG("Total number of bytes sent: %lu", total_number_of_bytes_sent_);
+    float average_bytes =
+        total_number_of_events_sent_ == 0
+            ? 0.0f
+            : static_cast<float>(total_number_of_bytes_sent_) / total_number_of_events_sent_;
```
Comment (dpallotti) [inline OrbitService/CaptureServiceImpl.cpp:100]: Dividing a float by zero doesn't produce any exception, it just gives NaN (technically more correct than 0...).

File: OrbitService/CaptureServiceImpl.cpp (hunk 2 above)
Comment (ronaldfw) [reply OrbitService/CaptureServiceImpl.cpp:94]: Done.

File: OrbitService/CaptureServiceImpl.cpp [hunk 4]
```diff
@@ -91,30 +91,52 @@ class GrpcCaptureEventSender final : public CaptureEventSender {
...
+    ORBIT_FLOAT("Average bytes per CaptureEvent", average_bytes);
+    total_number_of_events_sent_ += events.size();
+    total_number_of_bytes_sent_ += number_of_bytes_sent;
   }
 
  private:
   grpc::ServerReaderWriter<CaptureResponse, CaptureRequest>* reader_writer_;
+
+  uint64_t total_number_of_events_sent_ = 0;
+  uint64_t total_number_of_bytes_sent_ = 0;
```
[the remark this answers is not in the document]
Comment (ronaldfw) [reply OrbitService/CaptureServiceImpl.cpp:136]: No, I don't need it, but would prefer to keep it.

File: OrbitService/CaptureServiceImpl.cpp (hunk 3 above)
Comment (dpallotti) [reply OrbitService/CaptureServiceImpl.cpp]: (Well, NaN if the dividend is also 0, plus or minus infinity otherwise.)

Comment (ronaldfw) [reply OrbitService/CaptureServiceImpl.cpp]: Ah, of course. Was thinking of integer division for a moment. Reverted the change (which is also more concise). Thanks a lot!

File: OrbitService/CaptureServiceImpl.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (dpallotti) [reply OrbitService/CaptureServiceImpl.cpp:136]: Then I don't really understand the reasoning, but ok, it doesn't matter.

**Review by dpallotti — APPROVED** [review]

**Pull request merged**